sql/core/src/main/scala/org/apache/spark/sql/execution/joins/SortMergeJoinExec.scala

```diff
@@ -41,6 +41,42 @@ case class SortMergeJoinExec(
     left: SparkPlan,
     right: SparkPlan) extends BinaryExecNode with CodegenSupport {
 
+  lazy val (reorderedLeftKeys, reorderedRightKeys) = {
```

Comment:
I looked at #17339 and its doing something orthogonal to whats done here.

#17339 is ensuring that the join outputs' sort ordering has attributes from both relations. 
This PR is ensuring that the order of join kets (in both distribution and sort order) is not blindly picked from the order of occurrence in the query string.